profiles/views.py

```diff
@@ -172,22 +172,16 @@ def get_serializer_class(self):
                 if get_contacts
                 else ProfileDetailSerializer
             )
+        elif self.request.method == "DELETE":
+            return ProfileDeleteSerializer
         else:
             return ProfileOwnerDetailEditSerializer
 
-    def destroy(self, request, *args, **kwargs):
-        instance = self.get_object()
-        user = self.request.user
-        password = self.request.data.get("password")
-        if not password or not check_password(password, user.password):
-            return Response(status=status.HTTP_400_BAD_REQUEST)
-        else:
-            self.perform_destroy(instance)
-            return Response(status=status.HTTP_204_NO_CONTENT)
-
     def perform_destroy(self, instance):
-        instance.is_deleted = True
-        instance.save()
+        serializer = self.get_serializer(data=self.request.data)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
There is no need do put it into `if`. Just call `serializer.is_valid(raise_exception=True)`, if will raise exception and return 400 BAD REQUEST in case of validation issues anyway.

Reply:
Fixed